sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/xml/XmlInputFormat.scala

```diff
@@ -117,7 +117,10 @@ private[xml] class XmlRecordReader extends RecordReader[LongWritable, Text] {
       }
     } else {
       fsin.seek(start)
-      countingIn = new CountingInputStream(fsin)
+      countingIn = BoundedInputStream.builder()
+        .setInputStream(fsin)
+        .setMaxCount(fileSplit.getLength)
```

Comment:
Is this really necessary in the current scenario? Perhaps the `unbound` one is enough?

Reply:
Sounds good, changed to `unbound `.